### PR TITLE
Use g:hier_highlight_group_* as a highlight

### PR DESCRIPTION
**Pull request by uplus**
Hello.
I want to change highlights of hier with g:hier_highlight_group_*

### CODE REVIEW

File: plugin/hier.vim [hunk 1]
```diff
@@ -8,35 +8,13 @@ if (exists("g:loaded_hier") && g:loaded_hier) || &cp
...
-if eval('g:hier_highlight_group_loci') != ''
-	exec "hi! link LocInfo    ".g:hier_highlight_group_loci
-endif
+let g:hier_enabled = get(g:, 'hier_enabled', 1)
+let g:hier_highlight_group_qf  = get(g:, 'hier_highlight_group_qf', 'QFError')
+let g:hier_highlight_group_qfw = get(g:, 'hier_highlight_group_qfw', 'QFWarning')
+let g:hier_highlight_group_qfi = get(g:, 'hier_highlight_group_qfi', 'QFInfo')
+let g:hier_highlight_group_loc	= get(g:, 'hier_highlight_group_loc', 'LocError')
+let g:hier_highlight_group_locw = get(g:, 'hier_highlight_group_locw', 'LocWarning')
+let g:hier_highlight_group_loci = get(g:, 'hier_highlight_group_loci', 'LocInfo')
```
Comment (pocke) [inline plugin/hier.vim]: The six highlight groups are not defined by Vim's default. 
So, if the variables are not defined, this plugin crashes with the following message.

```
Error detected while processing function <SNR>129_Hier:
line   32:
E28: No such highlight group name: LocError
Error detected while processing function <SNR>196_receive_vimproc_result[26]..64:
line    9:
E171: Missing :endif
```


In original, `Spellbad`, `SpellLocal` and `SpellRare` are assigned into the variables by default. I think we use those highlight groups as default value.

File: plugin/hier.vim [hunk 2]
```diff
@@ -64,11 +42,11 @@ function! s:Hier(clearonly)
...
 		for i in s:Getlist(0, type)
 			if i.bufnr == bufnr
-				let hi_group = 'QFError'
+				let hi_group = eval('g:hier_highlight_group_'.type)
 				if i.type == 'I' || i.type == 'info'
-					let hi_group = 'QFInfo'
+					let hi_group = eval('g:hier_highlight_group_'.type.'i')
 				elseif i.type == 'W' || i.type == 'warning'
-					let hi_group = 'QFWarning'
+					let hi_group = eval('g:hier_highlight_group_'.type.'w')
```
Comment (pocke) [inline plugin/hier.vim]: Could you replace `eval()` with `get()` ?

See. https://twitter.com/pink_bangbi/status/810300975357591557

**Review by pocke — CHANGES_REQUESTED** [review]

**Comment by pocke** [comment]
Hi, thanks for your pull-request!

I made some comments. Could you check the comments?

**Comment by pocke** [comment]
Note:
I would like to propose that we merge this change, this repository and cohama/vim-hier to original repositoy (jceb/vim-hier) after this PR is merged.

File: plugin/hier.vim (hunk 1 above)
Comment (uplus) [reply plugin/hier.vim]: Sorry, I didn't know.
I fixing.

File: plugin/hier.vim (hunk 2 above)
Comment (uplus) [reply plugin/hier.vim]: Yes, I can.

**Comment by uplus** [comment]
That's a good idea.

**Review by pocke — APPROVED** [review]

**Comment by pocke** [comment]
It's good. Thanks!

**Pull request merged**